### PR TITLE
Add aten.tensor.int and aten.tensor.float op lowerings.

### PR DESCRIPTION
**Pull request by pashu123**
Add the required lowerings and correct test cases.
These op produce zero-d tensors and it was incorrectly mentioned in
refine types to produce 1d tensor of size 1.

### CODE REVIEW

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp [hunk 1]
```diff
@@ -3451,6 +3451,65 @@ class ConvertAtenContiguousOp : public OpConversionPattern<AtenContiguousOp> {
...
+      dtype = tensorIntOp.dtype();
+      device = tensorIntOp.device();
+      requires_grad = tensorIntOp.requires_grad();
+    }
+    if (AtenTensorFloatOp tensorFloatOp = dyn_cast<AtenTensorFloatOp>(op)) {
+      AtenTensorFloatOp::Adaptor adaptor(operands);
+      elemVal = adaptor.t();
+      dtype = tensorFloatOp.dtype();
+      device = tensorFloatOp.device();
+      requires_grad = tensorFloatOp.requires_grad();
```
Comment (Shukla-Gaurav) [inline lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:3486]: Since all the members are same in both the operations, you should templatize it.

**Review by Shukla-Gaurav — CHANGES_REQUESTED** [review]
Few comments here.

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp [hunk 2]
```diff
@@ -3451,6 +3451,65 @@ class ConvertAtenContiguousOp : public OpConversionPattern<AtenContiguousOp> {
...
+      return rewriter.notifyMatchFailure(op, "Unimplemented non-None dtype");
+
+    // TODO: Device information.
+    if (!device.getType().isa<Torch::NoneType>())
+      return rewriter.notifyMatchFailure(
+          op, "Unimplemented non-None device information");
+
+    RankedTensorType resultType = getTypeConverter()
+                                      ->convertType(op->getResult(0).getType())
+                                      .cast<RankedTensorType>();
```
[the remark this answers is not in the document]
Comment (pashu123) [reply lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:3499]: Complains `no member named 'getType' in 'mlir::Operation'.

[the remark this answers is not in the document]
Comment (cathyzhyi) [reply lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:3499]: did you use `op->getType()`?

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp (hunk 1 above)
Comment (pashu123) [reply lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:3486]: This will evolve in the future and may contain ops like `AtenTensorOp` where some of the members' names change. Hence fingers crossed.

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (pashu123) [reply lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:3499]: yes.

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp [hunk 3]
```diff
@@ -3451,6 +3451,63 @@ class ConvertAtenContiguousOp : public OpConversionPattern<AtenContiguousOp> {
 };
 } // namespace
 
+namespace {
+struct ConvertAtenScalarToTensorLike : ConversionPattern {
+  ConvertAtenScalarToTensorLike(TypeConverter &typeConverter,
+                                MLIRContext *context)
+      : ConversionPattern(typeConverter, MatchAnyOpTypeTag(), /*benefit=*/1,
+                          context) {}
+
```
Comment (Shukla-Gaurav) [inline lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:3460]: No need of extra blank space here.

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp [hunk 4]
```diff
@@ -3451,6 +3451,63 @@ class ConvertAtenContiguousOp : public OpConversionPattern<AtenContiguousOp> {
...
+    if (!isa<AtenTensorIntOp, AtenTensorFloatOp>(op))
+      return rewriter.notifyMatchFailure(
+          op, "not a supported Scalar to Tensor like op");
+
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op->getLoc();
+
+    Value elemVal, dtype, device, requires_grad;
+
```
Comment (Shukla-Gaurav) [inline lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:3473]: Please avoid extra blank spaces.

File: lib/Dialect/Torch/Transforms/RefineTypes.cpp [hunk 5]
```diff
@@ -238,7 +238,7 @@ class TypeAnalyzer : public ForwardDataFlowAnalysis<ValueKnowledge> {
             AtenGeluBackwardOp, AtenBitwiseNotOp, AtenExpOp, AtenSinOp,
             AtenCosOp, AtenSigmoidOp, DerefineOp, AtenToPrimDeviceOp, AtenCpuOp,
             AtenContiguousOp, AtenFill_ScalarOp, AtenDetachOp,
-            AtenMaskedFill_ScalarOp, AtenCopy_Op, AtenIndexPut_Op, AtenCumsumOp,
+            AtenMaskedFill_ScalarOp, AtenCopy_Op, AtenIndexPutOp, AtenCumsumOp,
```
Comment (Shukla-Gaurav) [inline lib/Dialect/Torch/Transforms/RefineTypes.cpp:241]: `AtenIndexPutOp` seems unrelated here(wrt the commit).

**Review by Shukla-Gaurav — APPROVED** [review]
LGTM! Few minor comments to handle. Thanks!
